tests/phpunit/CRM/Core/Payment/PayPalPNTest.php

```diff
@@ -68,6 +67,35 @@ public function tearDown() {
     $this->quickCleanUpFinancialEntities();
   }
 
+  public function testIPNPaymentSuccess() {
+    $pendingStatusID = CRM_Core_OptionGroup::getValue('contribution_status', 'Pending', 'name');
+    $completedStatusID = CRM_Core_OptionGroup::getValue('contribution_status', 'Completed', 'name');
+    $params = array(
+      'payment_processor_id' => $this->_paymentProcessorID,
+      'contact_id' => $this->_contactID,
+      'trxn_id' => NULL,
+      'invoice_id' => $this->_invoiceID,
+      'contribution_status_id' => $pendingStatusID,
+    );
+    $this->contributionCreate($params);
+    $contribution = $this->callAPISuccess('contribution', 'get', array('id' => $this->_contributionID, 'sequential' => 1));
+    // assert that contribution created before handling payment via paypal standard has no transaction id set and pending status
+    $this->assertEquals(NULL, $contribution['values'][0]['trxn_id']);
+    $this->assertEquals($pendingStatusID, $contribution['values'][0]['contribution_status_id']);
+
+    global $_REQUEST;
+    $_REQUEST = $this->getPaypalTransaction();
+
+    $paymentProcesors = civicrm_api3('PaymentProcessor', 'getsingle', array('id' => $this->_paymentProcessorID));
+    $payment = Civi\Payment\System::singleton()->getByProcessor($paymentProcesors);
+    $payment->handlePaymentNotification();
```

Comment:
cool - we haven't any coverage of calling the IPN this way so this extends coverage